### PR TITLE
Plotting upgrades

### PR DESCRIPTION
**Pull request by dcherian**
- [x] Tests added (for all bug fixes or enhancements)
 - [x] Tests passed (for all non-documentation changes)
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API (remove if this change should not be visible to users, e.g., if it is an internal clean-up, or if this is part of a larger project that will be documented later)

`plot.line()` now accepts `xincrease`, `yincrease` kwargs (see bottom panel of image).

Additionally, instead of calling `autofmt_xdate()` which deletes all subplot x-axes except the last one, I rotate and set alignment for the x-axes labels manually. This allows for creating figures where not all subplots have time on the x-axis.

image

### CODE REVIEW

File: xarray/plot/plot.py [hunk 1]
```diff
@@ -182,6 +182,12 @@ def line(darray, *args, **kwargs):
         Coordinates for x, y axis. Only one of these may be specified.
         The other coordinate plots values from the DataArray on which this
         plot method is called.
+    xincrease : None, True, or False, optional
+        Should the values on the x axes be increasing from left to right?
+        if None, use the default for the matplotlib function
```
Comment (shoyer) [inline xarray/plot/plot.py]: Please add a period at the end of the sentence.

File: xarray/plot/plot.py [hunk 2]
```diff
@@ -270,7 +278,11 @@ def line(darray, *args, **kwargs):
 
     # Rotate dates on xlabels
     if np.issubdtype(xplt.dtype, np.datetime64):
-        ax.get_figure().autofmt_xdate()
+        for xlabels in ax.get_xticklabels():
+            xlabels.set_rotation(30)
+            xlabels.set_ha('right')
```
Comment (shoyer) [inline xarray/plot/plot.py]: 1. Let's add a link to [this StackOverflow post](https://stackoverflow.com/questions/17430105/autofmt-xdate-deletes-x-axis-labels-of-all-subplots) in a comment.
2. Do we even need this datetime64 logic with the latest version of matplotlib? matplotlib 2.2.x added some sort of built-in support for datetime64: https://github.com/matplotlib/matplotlib/pull/9779.

Comment (dcherian) [reply xarray/plot/plot.py]: Also, I am on 2.2.2...

Comment (shoyer) [reply xarray/plot/plot.py]: OK, thanks for checking!

**Comment by dcherian** [comment]
This last commit prevents this warning:
```
"ResourceWarning: unclosed file <_io.BufferedReader name='/home/deepak/work/python/xarray/xarray/plot/default_colormap.csv'>
  cmap = _load_default_cmap()"
```

**Comment by shoyer** [comment]
thanks!

**Pull request merged**